File: src/easy/MajorityElement.go
package main

import (
	"fmt"
)

func majorityElement(nums []int) int {
	hashTable := make(map[int]int)

	for _, i := range nums {
		hashTable[i]++
	}

	for element, majority := range hashTable {
		if majority > len(nums)/2 {
			return element
		}
	}

	return 1
}

func main() {
	fmt.Println(majorityElement([]int{3, 2, 3}))
	fmt.Println(majorityElement([]int{2, 2, 1, 1, 1, 2, 2}))
}

File: src/easy/MajorityElement.java
public class MajorityElement {
    public int majorityElement(int[] nums) {
        int candidate = 0;
        int counter = 0;
        for (int num : nums) {
            if (counter == 0) {
                candidate = num;
            }

            if (candidate == num) {
                counter++;
            } else {
                counter--;
            }
        }
        return candidate;
    }
}
